src/index.ts

```diff
@@ -1,17 +1,18 @@
 import { Observable, Subscription, BehaviorSubject } from 'rxjs'
+import * as WebSocket from 'ws';
 
 export interface Connection {
   connectionStatus: Observable<number>,
-  messages: Observable<string>,
+  messages: Observable<WebSocket.Data>,
 }
 
 export interface IWebSocket {
-  close()
-  send(data: string | ArrayBuffer | Blob)
-  onopen?: (event: Event) => any
-  onclose?: (event: CloseEvent) => any
-  onmessage?: (event: MessageEvent) => any
-  onerror?: (event: ErrorEvent) => any
+  close(): void
+  send(data: WebSocket.Data)
+  onopen?: (event: { target: WebSocket }) => void;
```

Comment:
Taken from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/ws

Reply:
The `send` typings here are better though, I need to integrate that part of the change.